Updated local locale files, Fixed some fileio questions

Angels-D opened the pull request:
* I tried to do some translations and added some tools for translation, and the translations ❓ of the tools were tagged for further corrections
  > It is now possible to automatically generate some text that has not been added to the translation by uncommenting in translate()
  > Some of the existing translations are from the original json files, and some fields are no longer needed, but it is cumbersome to locate those that are not needed so they have not been removed

* The game is not friendly to GBK-encoded CFG documents, so it may be better to force UTF-8

I was going to submit a PR after all the translations were completed, but you were too diligent in updating👍
I'm sorry my English isn't very good (so I need a translator urgently)  ( ´･･)ﾉ(._.`)
I don't know how to submit a PR yet, I hope I can help you 😊

Sly0511 commented:
Hi, thank you for your Pull Request and willingness to help firstly.

I wanted to make a "better" approach for translating that didn't make it painful in dev environment, my previous approach wasnt further developed because it just wasn't pleasant to use in dev environment and was too confusing in locale files.

I will now review this PR in dev and see what issues may arise from it, thank you again I'll get back to you shortly

Sly0511 commented on models/constants.py:
Looking back at this, I am unsure why I didn't use Path object properly

```py
{
    path.as_posix(): load(
        locales_path.joinpath(path).read_text()
    )
    for path in files_data
}
 ````

Angels-D replied:
By default, the program prioritizes access to network resources, so it may be overlooked.
I'm removing some fields from the loc file that don't need to be translated, and some descriptive information about the star map that aren't showing up on the interface and that I accidentally added to the translation.

Sly0511 replied:
I can try and implement a local mode for translations as well

Sly0511 reviewed (changes requested):
Overall the only main issue I can find is that, when you are doing strings, you are taking advantage that Python 3.12 can now read double quotations in cascade without affecting it's functionality in f-strings, this however is not a present feature in Python 3.11.8 which is the version the executable/installer is compiled into, this means compilation will fail on next release.

If you could just fix this detail please

Thank you again for the great work and help in this PR

Angels-D commented:
I'm looking up what has changed to the f-string
such as `f"ABC{Dict["key"]}"` needs to be changed to  `f"ABC{Dict['key']}"` 
Right?

Sly0511 commented:
Correct!

Sly0511 commented:
There's another issue, as a personal preference, I dislike exposing newlines in translation files
![image](https://github.com/Sly0511/RenewedTroveTools/assets/47339373/0bfbb27c-a73a-40f1-9424-9adf8b2cb063)
I'd rather have a bit messier code, over broken strings in this case to avoid people not getting how translation files work in this regard.

Sly0511 commented:
Excuse my last comment, it seems to be a in game string haha sorry

Angels-D commented:
Haha, but I haven't figured out how to translate this yet
``` python
        task_lines = [
            "Performance mode is a very sensitive mode, it will create a cache of some of the file's data and improve "
            "performance of changes detection beyond imaginable",
            "But it comes with the caveat that it will have the wrong results if you use any other tool, "
            "this is because it won't have the information that an extraction or update happened, so when trying to "
            "track changes through this cache it will assume a state that may not match."
            "If you only plan to use this tool as your only extraction method, you may enable this with no worry"
            "If you plan on using any other extraction method however, be weary of the issues this cache may present "
            "to the accuracy of change tracking."
            "Even out of performance mode, this app will most likely manage faster speeds than other methods (I know "
            "of).",
        ]
```

Sly0511 commented:
task lines are later compiled into a string so you can do it there? however there's a way of doing this by recompiling

task_lines = [...]
task_lines = [loc(l) for l in task_lines]

Sly0511 commented:
I think we can do a trick in the Localization manager class where we can compile strings with new lines...and split them, translate one by one, rejoin them with new lines and return, this helps keep both loc file and code clean, I'll implement this myself

Sly0511 commented on tools/translation.py:
I am honestly fine with this we can use it

Angels-D commented:
Some of the automatic translations are terrible, and I think it's better to show the original text.
Removed a lot of translations about the description of the star chart, and those fields would not be displayed, so I deleted them, which I didn't pay attention to when I added fields in batches before.

Sly0511 commented:
Also if you wish you can either add me on discord through `sly.dev.` or join the app's discord, however since you are in china, I am clueless of the restrictions you face.

As for the translations of the star chart, I've contacted owner of Trovesaurus, I may be able to snatch from in game some of the strings, there's also a chinese mod which I can reach out for help as well

Sly0511 commented:
So far it looks good to me, if there are any changes you want to make, do them, else I can green light this patch into main branch and start working on some things, should beable to release these changes today

Sly0511 commented:
Thank you again, no rush the work you did was gold already, really appreciate the help.

I purposefully worked a few days ago the translation engine to make sure it was a bit less confusing to work with and more automatable.

I'll now close this.
